pkg/controller/controller.go

```diff
@@ -995,8 +999,10 @@ func (c *controller) pollInstance(serviceClass *v1alpha1.ServiceClass, servicePl
 				successDeprovisionMessage,
 			)
 			// Clear the finalizer
-			if len(instance.Finalizers) > 0 && instance.Finalizers[0] == "kubernetes" {
-				c.updateInstanceFinalizers(instance, instance.Finalizers[1:])
+			finalizers := sets.NewString(instance.Finalizers...)
```

Comment:
do we need finalizers outside of this couple of lines of context? Can we smash it into the if statement?

```
			if finalizers := sets.NewString(instance.Finalizers...); finalizers.Has(v1alpha1.FinalizerServiceCatalog) {
				finalizers.Delete(v1alpha1.FinalizerServiceCatalog)
				c.updateInstanceFinalizers(instance, finalizers.List())
			}
```
```